code/modules/mob/living/simple_animal/abnormality/he/golden_false_apple.dm

```diff
@@ -311,15 +285,29 @@
 	is_maggot = TRUE
 	SpeedChange(-1)
 
-/mob/living/simple_animal/hostile/abnormality/golden_apple/AttackingTarget()
+/mob/living/simple_animal/hostile/abnormality/golden_apple/AttackingTarget()//regular attacks or AOE. Determines the outcome for both players and the AI behavior
 	if(!can_act)
 		return FALSE
-	if(!is_maggot)
+	if(!is_maggot)//Is it still in the first form? Start building sheen pulses
+		if(pulse_count < pulse_maximum)
+			if(isliving(target))
+				var/mob/living/hit = target
+				if((hit.stat == DEAD) ||!ishuman(hit))//if the target is dead or not human
+					return ..()
+				pulse_count += 1
 		return ..()
-	if(prob(50))
-		return ..()
-	Smash(target)
-	return
+	if(client && smash_cooldown < world.time)//playable behavior is nested under here
+		switch(chosen_attack)
+			if(1)
+				return ..()
+			if(2)
+				Smash(target)
+			if(3)
+				Smash(target, wide = FALSE)
+		return
+	if(prob(50) && (smash_cooldown < world.time))//AI behavior goes here
+		(Smash(target, wide = pick(TRUE, FALSE)))
+	return ..()
```

Comment:
Note that by running Smash and `..()` you hit the main target *twice*.

Reply:
Fixed in the next push